### ISSUE TITLE
Unexpected quit on macOS 10.14 beta

### ISSUE THREAD

**Issue by chinnnki** [mpv-player/mpv#5908]
### mpv version and platform

HEAD-4d2b865 (compiled via Homebrew)
macOS 10.14 ~~beta (18A293u)~~ betas

### Reproduction steps

Open any video file with mpv bundle, or just simply open the bundle.

### Expected behavior

Normal playback.

### Actual behavior

The bundle quits unexpectedly.

### Log file

Unexpected quit with HEAD-4d2b865: https://0x0.st/s_xO.txt
Normal playback with ~~0.28.0~~ pre-bundled 0.28.2: https://0x0.st/s_xp.txt

**Comment by Akemi** [comment]
if you can use a debugger (lldb) and tell me where it is stuck that would be helpful. otherwise you will have to wait till i feel like installing the beta or it is released.

**Comment by chinnnki** [comment]
I'd like to give it a shot, however my limited coding knowledge will likely keep me from getting it done.
If it's not too much trouble, could you kindly tell me where to begin?
Also, would the system crash report be helpful?

**Comment by Akemi** [comment]
yes if you get a system crash report that is basically what i am looking for.

assuming that you build mpv yourself, since you tested current master, you can try PR #5753. it might fix your problem because it changed the initialising code.

**Comment by chinnnki** [comment]
Here is the crash report: https://0x0.st/s_3F.txt.

Thanks! I will try that.

**Comment by Akemi** [comment]
it crashes at a very unexpected point. nothing really changed there between 0.28 and master. weird that it crashes with the latter.

thx for the crash report.

**Comment by chinnnki** [comment]
Thanks!

BTW, tried PR #5753. It worked.

**Comment by Akemi** [comment]
okay. that is a bit unexpected too, but i might be forgetting something. anyway good that the PR fixes it for you.

**Comment by chinnnki** [comment]
Sorry for bothering again.

After some testing, I began to think this was all Homebrew's fault. I tried building current master using waf without PR #5753, and the playback was perfectly normal. The issue only occurs when playing a file with the .app bundle that comes with the `brew --with-bundle` compilation.

**Comment by Akemi** [comment]
PR #5753 + bundle are fine still?

**Comment by chinnnki** [comment]
Correction:

Currently, using command line,

- Current master (manually or `brew` compiled) works fine when `vo=gpu`, but crashes a lot when `vo=libmpv` (ending with a `Segmentation fault: 112 (0%) A-V:  0.000 Dropped: 1` alert. Not sure what that means).
- 0.28.2 (`brew` compiled) opens, but there's no video image.
- PR #5753 works fine with both `vo`.

As for bundles, they all keep crashing (window simply doesn't show up).

However, before upgrading to 10.14 beta, I had been using a `brew` compiled 0.28.2 + bundle. This one still works fine in current 10.14, both command line and bundle.

**Comment by zviratko** [comment]
mpv stopped working for me too (homebrew version)
it quit if given any filename with
```
Assertion failed: (self->_mk_tap_port != nil), function -[EventsResponder startMediaKeys]_block_invoke, file ../osdep/macosx_events.m, line 407.
Abort trap: 6
```

this was solved by --input-media-keys=no

then it starts playing, but there's something like a black overlay over the video - when I resize the window the video "flickers through"

**Comment by Akemi** [comment]
yes the flickering with vo=gpu, the old default, was a know issue even before. it's not fixable (like many other things), hence why i deprecate it and vo=libmpv replaced it. good that the newest version works for you.

i will look at the bundle problem mentioned before whenever i can install 10.14. i believe it's not in open beta yet.

**Comment by chinnnki** [comment]
Wow, vo=gpu was deprecated? Didn't notice that. I thought only opengl-cb was deprecated.

Anyway, thanks for the help! 
BTW, public beta is due later this month.

**Comment by Hrxn** [comment]
Uhm.. no?

vo=gpu is only deprecated for macOS, as far as I know, so this only applies to this OS.

**Comment by Akemi** [comment]
yeah, i meant vo=gpu with gpu-context=cocoa. i didn't think i had to be that explicit about it since this issue is only macOS related.

**Comment by zviratko** [comment]
Another day.
Another macOS dev beta update.
Another bug.

`2018-06-19 23:10:42.833 mpv[20152:192852] *** Terminating app due to uncaught exception 'NSImageCacheException', reason: 'Cannot lock focus on image <NSImage 0x7f9cdeebea40 Size={0, 0} Reps=(
)>, because it is size zero.'
*** First throw call stack:
(
	0   CoreFoundation                      0x00007fff42066484 __exceptionPreprocess + 197
	1   libobjc.A.dylib                     0x00007fff6e191d08 objc_exception_throw + 48
	2   CoreFoundation                      0x00007fff420e84f9 +[NSException raise:format:] + 201
	3   AppKit                              0x00007fff3f6b894d -[NSImage _lockFocusOnRepresentation:rect:context:hints:flipped:] + 316
	4   AppKit                              0x00007fff3f6b8808 __51-[NSImage lockFocusWithRect:context:hints:flipped:]_block_invoke + 68
	5   AppKit                              0x00007fff3f683626 -[NSImage _usingBestRepresentationForRect:context:hints:body:] + 156
	6   AppKit                              0x00007fff3f6b87ba -[NSImage lockFocusWithRect:context:hints:flipped:] + 141
	7   AppKit                              0x00007fff3f71706f -[NSImage lockFocusFlipped:] + 111
	8   AppKit                              0x00007fff3f88a921 -[NSApplication _setApplicationIconImage:setDockImage:] + 273
	9   mpv                                 0x00000001083f6aca $S11macOS_swift7CocoaCBC11initBackendyySpySo2voVGF + 330
	10  mpv                                 0x0000000108409747 $S11macOS_swift7CocoaCBC8reconfigyySpySo2voVGFyyXEfU_TA + 23
	11  mpv                                 0x0000000108401c41 $SIg_Ieg_TRTA + 17
	12  mpv                                 0x000000010840b579 $SIg_Ieg_TRTA.68 + 9
	13  mpv                                 0x00000001083ef9b0 $SIeg_IeyB_TR + 32
	14  libdispatch.dylib                   0x00007fff6f0d99a6 _dispatch_client_callout + 8
	15  libdispatch.dylib                   0x00007fff6f0f1063 _dispatch_async_and_wait_invoke + 87
	16  libdispatch.dylib                   0x00007fff6f0d99a6 _dispatch_client_callout + 8
	17  libdispatch.dylib                   0x00007fff6f0e47e4 _dispatch_main_queue_callback_4CF + 1129
	18  CoreFoundation                      0x00007fff4202cfbb __CFRUNLOOP_IS_SERVICING_THE_MAIN_DISPATCH_QUEUE__ + 9
	19  CoreFoundation                      0x00007fff41ff7be9 __CFRunLoopRun + 2341
	20  CoreFoundation                      0x00007fff41ff70a7 CFRunLoopRunSpecific + 463
	21  HIToolbox                           0x00007fff4128914b RunCurrentEventLoopInMode + 293
	22  HIToolbox                           0x00007fff41288e85 ReceiveNextEventCommon + 618
	23  HIToolbox                           0x00007fff41288c02 _BlockUntilNextEventMatchingListInModeWithFilter + 64
	24  AppKit                              0x00007fff3f57519e _DPSNextEvent + 1000
	25  AppKit                              0x00007fff3fcc0aa6 -[NSApplication(NSEvent) _nextEventMatchingEventMask:untilDate:inMode:dequeue:] + 1356
	26  AppKit                              0x00007fff3f56a88e -[NSApplication run] + 699
	27  mpv                                 0x00000001083dfd6c cocoa_main + 1052
	28  libdyld.dylib                       0x00007fff6f114b11 start + 1
	29  ???                                 0x0000000000000002 0x0 + 2
)
libc++abi.dylib: terminating with uncaught exception of type NSException
Abort trap: 6`

Any ideas? How can I help debug this?

version HEAD-cd89362

**Comment by Akemi** [comment]
seems like it crashes [here](https://github.com/mpv-player/mpv/blob/master/video/out/cocoa_cb_common.swift#L100). more specifically [here](https://github.com/mpv-player/mpv/blob/master/video/out/cocoa_cb_common.swift#L141) and in this [method](https://github.com/mpv-player/mpv/blob/master/osdep/macosx_application.m#L125). the image size is zero for whatever reason. only thing to do is to find out why the icon isn't returned properly.

i wonder when the public beta starts. though it might be best to wait for it to mature a bit more, maybe those bugs are just temporary.

**Comment by zviratko** [comment]
Sure, no need to spend time when things are still in flux, maybe it will start working by the next beta :) 

In the meantime should I try just commenting out setAppIcon to see if it works, or is there another possible workaround?

**Comment by Akemi** [comment]
yes commenting [setAppIcon](https://github.com/mpv-player/mpv/blob/master/video/out/cocoa_cb_common.swift#L100) should work.

**Comment by zviratko** [comment]
Commenting it out worked.
Thanks.

**Comment by Tsukimeizi** [comment]
10.14 public beta is released, and I get the same issue.

Playing: [SumiSora&MAGI_ATELIER&CASO][Fate_stay_night_Heaven's_Feel][01][GB][720p].mp4
 (+) Video --vid=1 (*) (h264 1280x720 23.976fps)
 (+) Audio --aid=1 --alang=jpn (*) (aac 2ch 48000Hz)
Using hardware decoding (videotoolbox).
AO: [coreaudio] 48000Hz stereo 2ch floatp
VO: [libmpv] 1280x720 videotoolbox[nv12]
2018-06-27 16:32:10.284 mpv[45164:280931] *** Terminating app due to uncaught exception 'NSImageCacheException', reason: 'Cannot lock focus on image <NSImage 0x7fd4f843fc00 Size={0, 0} Reps=(
)>, because it is size zero.'
*** First throw call stack:
(
	0   CoreFoundation                      0x00007fff32bef484 __exceptionPreprocess + 197
	1   libobjc.A.dylib                     0x00007fff5ed1ad08 objc_exception_throw + 48
	2   CoreFoundation                      0x00007fff32c714f9 +[NSException raise:format:] + 201
	3   AppKit                              0x00007fff3024194d -[NSImage _lockFocusOnRepresentation:rect:context:hints:flipped:] + 316
	4   AppKit                              0x00007fff30241808 __51-[NSImage lockFocusWithRect:context:hints:flipped:]_block_invoke + 68
	5   AppKit                              0x00007fff3020c626 -[NSImage _usingBestRepresentationForRect:context:hints:body:] + 156
	6   AppKit                              0x00007fff302417ba -[NSImage lockFocusWithRect:context:hints:flipped:] + 141
	7   AppKit                              0x00007fff302a006f -[NSImage lockFocusFlipped:] + 111
	8   AppKit                              0x00007fff30413921 -[NSApplication _setApplicationIconImage:setDockImage:] + 273
	9   mpv                                 0x00000001032eaaca $S11macOS_swift7CocoaCBC11initBackendyySpySo2voVGF + 330
	10  mpv                                 0x00000001032fd747 $S11macOS_swift7CocoaCBC8reconfigyySpySo2voVGFyyXEfU_TA + 23
	11  mpv                                 0x00000001032f5c41 $SIg_Ieg_TRTA + 17
	12  mpv                                 0x00000001032ff579 $SIg_Ieg_TRTA.68 + 9
	13  mpv                                 0x00000001032e39b0 $SIeg_IeyB_TR + 32
	14  libdispatch.dylib                   0x00007fff5fc629a6 _dispatch_client_callout + 8
	15  libdispatch.dylib                   0x00007fff5fc7a063 _dispatch_async_and_wait_invoke + 87
	16  libdispatch.dylib                   0x00007fff5fc629a6 _dispatch_client_callout + 8
	17  libdispatch.dylib                   0x00007fff5fc6d7e4 _dispatch_main_queue_callback_4CF + 1129
	18  CoreFoundation                      0x00007fff32bb5fbb __CFRUNLOOP_IS_SERVICING_THE_MAIN_DISPATCH_QUEUE__ + 9
	19  CoreFoundation                      0x00007fff32b80be9 __CFRunLoopRun + 2341
	20  CoreFoundation                      0x00007fff32b800a7 CFRunLoopRunSpecific + 463
	21  HIToolbox                           0x00007fff31e1214b RunCurrentEventLoopInMode + 293
	22  HIToolbox                           0x00007fff31e11e85 ReceiveNextEventCommon + 618
	23  HIToolbox                           0x00007fff31e11c02 _BlockUntilNextEventMatchingListInModeWithFilter + 64
	24  AppKit                              0x00007fff300fe19e _DPSNextEvent + 1000
	25  AppKit                              0x00007fff30849aa6 -[NSApplication(NSEvent) _nextEventMatchingEventMask:untilDate:inMode:dequeue:] + 1356
	26  AppKit                              0x00007fff300f388e -[NSApplication run] + 699
	27  mpv                                 0x00000001032d3d6c cocoa_main + 1052
	28  libdyld.dylib                       0x00007fff5fc9db11 start + 1
)
libc++abi.dylib: terminating with uncaught exception of type NSException
Abort trap: 6

**Comment by Akemi** [comment]
apparently my MacBook Pro from mid 2010 isn't supported anymore, so i can't test or fix this (for now).

**Comment by Akemi** [comment]
yeah that was my fault. i will fix it up tomorrow. if you remove the last commit by me it should work properly again.

**Comment by Akemi** [comment]
#5953 should be fine and working properly again.

**Comment by sspilleman** [comment]
You can install [iina](https://lhc70000.github.io/iina/) (brew cask install iina). Its mpv based and working on mojave without a flaw....

**Comment by Akemi** [comment]
stop polluting our issues with unrelated bullshit.

**Comment by sspilleman** [comment]
I guess you're the friendly one @Akemi.... It is related. From the conversations it looks like the issue is fixed for some time now, but there isn't a working version still. I'm merely trying to point to a solution that people could use in the meantime until the PRs are merged.

**Comment by Akemi** [comment]
it's not a solution, especially since iina doesn't work flawlessly. i wouldn't have said anything if you just had posted it once here.

**Comment by csdesigns** [comment]
the last thing iina is, is flawless. maybe for a few streams that you have, but it fails on way more streams than mpv does. and I say this as someone who is really pulling for that project

**Comment by vjk3** [comment]
Had this issue in Mojave just now. I had mpv installed via `brew cask` and after Mojave install, it would open, then quit immediately. One of those crashes however threw up a message about mpv requiring Accessibility permissions so I tried enabling that... **it works!**

System Preferences ➞ Security & Privacy (pane) ➞ Privacy (tab) ➞ Accessibility ➞ (choose mpv in this list). You will need to enter your password to unlock the ability to edit this pane.

Hope this was helpful. Cheers 👍

**Comment by warrenwyf** [comment]
> Had this issue in Mojave just now. I had mpv installed via `brew cask` and after Mojave install, it would open, then quit immediately. One of those crashes however threw up a message about mpv requiring Accessibility permissions so I tried enabling that... **it works!**
> 
> System Preferences ➞ Security & Privacy (pane) ➞ Privacy (tab) ➞ Accessibility ➞ (choose mpv in this list). You will need to enter your password to unlock the ability to edit this pane.
> 
> Hope this was helpful. Cheers 👍

It works

**Comment by Akemi** [comment]
yeah the event tap needs those permissions since 10.14, it's also mentioned in PR #5953 and commit https://github.com/mpv-player/mpv/pull/5953/commits/5b42be97e3d6efda91c5a87ecd828a5a1051bbbc.

**Comment by zmwangx** [comment]
FYI opening files with an app bundle (in my case not even the app bundle provided by mpv, just an automator app running a shell script that launches mpv with certain computed arguments) is fine if accessibility privilege is granted, but using mpv from the command line doesn't seem to benefit from the setting:

```console
$ mpv ...
Assertion failed: (self->_mk_tap_port != nil), function -[EventsResponder startMediaKeys]_block_invoke, file ../osdep/macosx_events.m, line 413.
[1]    3828 abort      mpv
```

Pretty annoying.

**Comment by Akemi** [comment]
that's exactly why that PR is open. it was mentioned several time that it fixes exactly that issue.

**Comment by Akemi** [comment]
from the patch you linked in the commit body.

> System Preferences > Security & Privacy > Privacy > Accessibility and
> add mpv or your **terminal App** to the list.

**Comment by zmwangx** [comment]
Hmm, are you saying it should work *without the patch* if I give mpv and my terminal app accessibility permissions? Because I already did:

<img width="352" alt="screen shot 2018-09-26 at 3 50 35 pm" src="https://user-images.githubusercontent.com/4149852/46105448-ff308500-c1a3-11e8-9640-62e3d7edd3a7.png">

Anyway, living with the patch is okay for me.

**Comment by Akemi** [comment]
yep it did when i tested it, though i only tested with an older beta and not the release of 10.14.

**Comment by zmwangx** [comment]
Okay, I removed and reauthorized iTerm.app, then it worked. Maybe every new version needs to be reauthorized (sounds crazy), or something.

**Comment by babadoctor** [comment]
really easy fix you guys, it's because you need to allow terminal in accessibility 
<img width="633" alt="screen shot 2018-10-02 at 10 35 40 am" src="https://user-images.githubusercontent.com/8715318/46366016-f110b780-c62e-11e8-9fa1-636b81653d49.png">

**Comment by Akemi** [comment]
fixed via ada4f7c6006fc60189987b3ee012e7f4ced9926e and 1842a932d37638c910f4471d85e81554076c3d2c.
